themes/dotmatrix/dotmatrix.theme

```diff
@@ -0,0 +1,1053 @@
+                __dotmatrix_make_sep_soft ${DOTMATRIX_COLOR_SEP_LITE[@]+"${DOTMATRIX_COLOR_SEP_LITE[@]}"}
+                _DOTMATRIX_VCS+="${dotmatrix_sep}${dotmatrix_field}"
+            fi
+            if [[ -n "$head" ]] ; then
+                __dotmatrix_make_field "$head" ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+                __dotmatrix_make_sep_soft ${DOTMATRIX_COLOR_SEP_LITE[@]+"${DOTMATRIX_COLOR_SEP_LITE[@]}"}
+                _DOTMATRIX_VCS+="${dotmatrix_sep}${dotmatrix_field}"
+            fi
+        else # no stash
+            if [[ -n "$untracked" ]] ; then
+                __dotmatrix_make_field "$untracked" ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+                _DOTMATRIX_VCS+="${dotmatrix_field}"
+                if [[ -n "$head" ]] ; then
+                    __dotmatrix_make_field "$head" ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+                    __dotmatrix_make_sep_soft ${DOTMATRIX_COLOR_SEP_LITE[@]+"${DOTMATRIX_COLOR_SEP_LITE[@]}"}
+                    _DOTMATRIX_VCS+="${dotmatrix_sep}${dotmatrix_field}"
+                fi
+            else # no untracked
+                if [[ -n "$head" ]] ; then
+                    __dotmatrix_make_field "$head" ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+                    _DOTMATRIX_VCS+="${dotmatrix_field}"
+                fi
+            fi
+        fi # stash
+        return 1
+    else
+        _DOTMATRIX_VCS=""
+        return 0
+    fi # find_vcs
+}
+
+########################################################################
+# Helper functions
+########################################################################
+
+__dotmatrix_join_fields() {
+    dotmatrix_join_fields=""
+    local sep="${1-}"
+    shift
+    if [[ -n "$@" ]] ; then
+        local first_field="${1-}"
+        shift
+        dotmatrix_join_fields="${first_field}"
+        for f in "$@"; do
+            dotmatrix_join_fields+="${sep}${f}"
+        done
+    fi
+}
+
+__dotmatrix_make_field() { # string, (color array)
+    local lp_terminal_format string="${1-}"
+    [[ -z $string ]] && return 1
+
+    shift
+    local format="$@"
+    if [[ -z $format ]] ; then
+        lp_terminal_format -1
+    else
+        lp_terminal_format $format
+    fi
+
+    dotmatrix_field="${lp_terminal_format}${DOTMATRIX_SPACE_FIELD}${string}${DOTMATRIX_SPACE_FIELD}"
+}
+
+__dotmatrix_make_sep_plug() { # color
+    # fg, bg, bold, underline, fallback_fg, fallback_bg
+    #  0 = set last color to current color
+    # -1 = set color to last color
+    # -2 = keep last color
+    # -3 = set last fg to last bg (and conversely)
+    local lp_terminal_format sep
+
+    local bg="${2-}"
+    [[ -z $bg ]] && bg=-1
+    local fbg="${6-}"
+    [[ -z $fbg ]] && fbg=-1
+
+    sep="${DOTMATRIX_PLUG}"
+    if [[ "${_lp_last_ab_color-}" == "$bg" ]] ; then
+        sep="${DOTMATRIX_MEAN}"
+    fi
+
+    lp_terminal_format -3 $bg 0 0 -1 $fbg
```

Comment:
Again, these need to be quoted. Try setting `IFS` to something like `$'\n\r \t1'` and see how much of this code breaks. (I tested it, it's a lot.)